### PR TITLE
Rotor Position Control

### PR DESCRIPTION
**Pull request by dzalkind**
## Description and Purpose

Add open loop control mode where the azimuthal position of the rotor can be controlled using the generator torque.

Some other related changes
- Each blade can be controlled in open loop individually
- The output sampling period of ROSCO dbg files can be changed, similar to OpenFAST's `DT_Out`
- Added PID control function
- Move all `CntrPar%*` assignments into `ReadControlParameterFileSub`, fixing a checkpoint bug

## Type of change
What types of change is it?
_Select the appropriate type(s) that describe this PR_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [ ] Maintenance update
- [ ] Other (please describe)

TODO Items General:
- [x] Add example/test for new feature
- [x] Update registry
- [ ] Run testing

TODO Items API Change:
- [ ] Update docs with API change
- [x] Run update_rosco_discons.py in Test_Cases/
- [ ] Update DISCON schema

### CODE REVIEW

**Comment by dzalkind** [comment]
[comp_outputs.pdf](https://github.com/NREL/ROSCO/files/12197776/comp_outputs.pdf)
Testing looks okay compared to the previous version.  Slight difference at initialization and due to unsteady aero.

File: Examples/25_rotor_position_control.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+os.makedirs(example_out_dir,exist_ok=True)
+
+if platform.system() == 'Windows':
+    lib_name = os.path.realpath(os.path.join(this_dir, '../ROSCO/build/libdiscon.dll'))
+elif platform.system() == 'Darwin':
+    lib_name = os.path.realpath(os.path.join(this_dir, '../ROSCO/build/libdiscon.dylib'))
+else:
+    lib_name = os.path.realpath(os.path.join(this_dir, '../ROSCO/build/libdiscon.so'))
+
+def main():
```
Comment (abhineet-gupta) [inline Examples/25_rotor_position_control.py:32]: Ran into an error
image

**Review by abhineet-gupta — COMMENTED** [review]
Ran into an error, possibly because I had to use python 3.10 (instead of 3.8) as wisdem is only available for 3.9 and 3.10 on conda. Can discuss more.

File: Test_Cases/IEA-15-240-RWT-UMaineSemi/IEA-15-240-RWT-UMaineSemi_MoorDyn.dat
[the remark this answers is not in the document]
Comment (dzalkind) [reply Test_Cases/IEA-15-240-RWT-UMaineSemi/IEA-15-240-RWT-UMaineSemi_MoorDyn.dat:1]: No, I was getting failures in the cable control case, and this is a leftover debug attempt

File: Examples/25_rotor_position_control.py (hunk 1 above)
Comment (dzalkind) [reply Examples/25_rotor_position_control.py:32]: Update docs to use python=3.9, but we might be able to get rid of that exact designation.

**Pull request merged**